### PR TITLE
Update fsi.fs

### PR DESCRIPTION
**Pull request by ingted**
Last try if FSharpEnvironment.tryCurrentDomain() returns null value

### CODE REVIEW

**Review by cartermp — APPROVED** [review]
I think the change looks good, but @KevinRansom would appreciate your take

File: src/fsharp/fsi/fsi.fs
Comment (teo-tsirpanis) [inline src/fsharp/fsi/fsi.fs:2763]: Why not use `AppContext.BaseDirectory`?

Comment (ingted) [reply src/fsharp/fsi/fsi.fs:2763]: When using Python.net or native .net core hosting scenario to load DLR libraries, it seems AppContext.BaseDirectory doesn't work... (include configuration manager, app.config...)

Comment (ingted) [reply src/fsharp/fsi/fsi.fs:2763]: https://docs.microsoft.com/en-us/dotnet/core/tutorials/netcore-hosting

**Comment by ingted** [comment]
Excuse me, will this be merged? If not, is it possible to have another change and getting it to be merged?

File: src/fsharp/fsi/fsi.fs
Comment (teo-tsirpanis) [inline src/fsharp/fsi/fsi.fs:2765]: You could directly use `Path.GetDirectoryName` here; there is no reason to create a `FileInfo`.

**Comment by cartermp** [comment]
> will this be merged?

We need 2+ review approvals to merge most PRs. So this will have to have another team member look at it.

File: src/fsharp/fsi/fsi.fs
Comment (ingted) [reply src/fsharp/fsi/fsi.fs]: Thanks! I changed it!!

**Comment by ingted** [comment]
Hi @cartermp,

Sorry that I am just wondering if there will be another one reviewing this change...
If no, I think I should close it and figure out another workaround.

File: src/fsharp/fsi/fsi.fs
Comment (ingted) [reply src/fsharp/fsi/fsi.fs]: Oops, the new commit is created, however the PR hasn't yet updated.... should I close it and recreate a new PR?

**Comment by dsyme** [comment]
Which bug is this fixing please?

**Comment by vzarytovskii** [comment]
> Which bug is this fixing please?

I believe that's the one:
https://github.com/dotnet/fsharp/issues/11293

File: src/fsharp/fsi/fsi.fs
Comment (dsyme) [inline src/fsharp/fsi/fsi.fs:2759]: If the problem is that `FSharpEnvironment.tryCurrentDomain()` returns a `null` then please write something like this:

```fsharp
    let defaultFSharpBinariesDir = 
        let fallback() =
            let d = Assembly.GetExecutingAssembly()
            Path.GetDirectoryName d.Location
        match FSharpEnvironment.tryCurrentDomain() with
        | null -> fallback()
        | dom -> 
            match FSharpEnvironment.BinFolderOfDefaultFSharpCompiler(dom) with
            | None -> fallback()
            | Some dir -> dir

```

Comment (ingted) [reply src/fsharp/fsi/fsi.fs]: Done.

**Comment by ingted** [comment]
Hi guys, will this be reviewed recently?

**Review by vzarytovskii — APPROVED** [review]

**Comment by dsyme** [comment]
@KevinRansom I'm OK with this, but I see yo self-assigned, so we'll wait for your review

**Comment by KevinRansom** [comment]
The thing is all right, the code is in the wrong place, I will push an update.


Sent from Outlook<http://aka.ms/weboutlook>

________________________________

**Comment by ingted** [comment]
Hi @dsyme  and @KevinRansom,

I am sorry that I missed a test... could I add a test here: tests/fsharp/core?

Could I create an fsi-initialization folder something?

(Or Kevin will add it...)

Please let me know if there is anything I could do for help. ^_^ Thank you!

**Comment by KevinRansom** [comment]
@ingted, sorry things have taken so long, I have been focussed on other things for quite a while. Anyway, I'm looking at this again now.

**Comment by KevinRansom** [comment]
I can't push to a forks main.  So this PR will be managed here:  https://github.com/dotnet/fsharp/pull/11824

**Pull request closed without merging**
